src/org/daisy/dotify/formatter/impl/volume/TocSequenceEventImpl.java

```diff
@@ -121,16 +124,23 @@ public BlockSequence getBlockSequence(FormatterContext context, DefaultContext v
 					getSequenceProperties());
 			fsm.appendGroup(getTocStart(vars));
 			if (getRange()==TocProperties.TocRange.VOLUME) {
-				Collection<Block> volumeToc = data.filter(refToVolume(vars.getCurrentVolume(), crh));
+				int currentVolume = vars.getCurrentVolume();
+				Collection<Block> resumedBlocks = data.filterEntryOnResumed(rangeToVolume(currentVolume, crh));
+				Collection<Block> volumeToc = data.filterEntry(refToVolume(currentVolume, crh));
+				if (resumedBlocks.isEmpty() && volumeToc.isEmpty()) {
+					return null;
+				}
+				if (!resumedBlocks.isEmpty()) {
+					fsm.appendGroup(resumedBlocks);
+				}
 				if (!volumeToc.isEmpty()) {
 					fsm.appendGroup(volumeToc);
-				} else {
-					return null;
 				}
 			} else if (getRange()==TocProperties.TocRange.DOCUMENT) {
 				for (int vol = 1; vol <= crh.getVolumeCount(); vol++) {
-					Collection<Block> volumeToc = data.filter(refToVolume(vol, crh));
-					if (!volumeToc.isEmpty()) {
+					Collection<Block> resumedBlocks = data.filterEntryOnResumed(rangeToVolume(vol, crh));
+					Collection<Block> volumeToc = data.filterEntry(refToVolume(vol, crh));
```

Comment:
Just do one filtering. That will ensure the order of blocks as defined in the OBFL is met, and no content is duplicated.

Reply:
Done.

Reply:
Done.